### PR TITLE
Added support for .isolate() -- runs downstream operators in a separate PE.

### PR DESCRIPTION
**Pull request by wmarshall484**
Part of enhancement #70

### CODE REVIEW

**Comment by wmarshall484** [comment]
test this

**Comment by wmarshall484** [comment]
test this

**Comment by wmarshall484** [comment]
test this

**Comment by wmarshall484** [comment]
test this

File: java/src/com/ibm/streamsx/topology/TStream.java [hunk 1]
```diff
@@ -507,6 +507,32 @@
...
+     * ------transform1---|                           |--.isolate()--filter---
+     *                    |--.isolate()---transform3--|
+     * </code></pre>
+     * 
+     * Three separate processes will be created -- one process will contain
+     * transform1, another will contain both transform2 and transform3, and the
+     * last will contain filter.
+     * 
+     * <br><br>
+     * Merging an isolated stream with its parent stream will throw an exception.
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/TStream.java]: What is "merging" ?

File: java/src/com/ibm/streamsx/topology/TStream.java [hunk 2]
```diff
@@ -507,6 +507,32 @@
      *             than 1.0.
      */
     TStream<T> sample(double fraction);
+    
+    /**
+     * Return a TStream that runs in a separate process from the parent. The
```
Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/TStream.java]: parent/child hasn't been used to describe stream relationships, so it's not clear what defines a parent-child relationship. I'm not sure we need to define such a relationship either.

I think all that isolate() defines is that any processing applied to the returned stream will be in a separate process to the processing that produced this stream.

Where there are no specific instructions from the application, the Java Application API is free to decide the layout of a topology to PEs, thus it's not guaranteed that a stream's "children" will be in the the same PE.

Comment (ddebrunner) [inline java/src/com/ibm/streamsx/topology/TStream.java]: Good to comment that this only applies for distributed.

**Pull request merged**